src/mgr/MgrStandby.cc

```diff
@@ -277,8 +277,26 @@ void MgrStandby::send_beacon()
   }
 
   // Whether I think I am available (request MgrMonitor to set me
-  // as available in the map)
-  bool available = active_mgr != nullptr && active_mgr->is_initialized();
+  // as available in the map).
+  //
+  // The active mgr is marked available if:
+  // 1. The mon has chosen a standby to be active
+  // 2. The chosen active mgr has all of its modules initialized
+  //
+  // In extreme cases, if modules take very long to initialize (several retries are
+  // allowed; see "mgr_module_load_max_retries"), we will proceed to mark the chosen
+  // active mgr "available" to unblock other mgr functionality such as reporting PG
+  // availability. If this happens, a health error will be issued indicating which
+  // mgr modules got stuck initializing (See src/mgr/PyModuleRegistry.cc). This unblocks
+  // the rest of the mgr's functionality while making it clear that some modules
+  // are unusuable.
+  bool available = false;
+  if (active_mgr != nullptr) {
+    available = active_mgr->is_initialized() || active_mgr->exceeded_initialization_retires();
```

Comment:
I wonder if there is any inconsistency or potential for problems that we're marking the Mgr `available` but internally it's still `initializing`?

Reply:
Ah, do you mean that `initialized` and `initializing` booleans should be set to "True" and "False" respectively at this point? If so, I agree they should be set if we are declaring the mgr active. Let me know.